src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -2363,7 +2363,8 @@
         ;; the "next update" and "crl-number" will get updated by the process that
         ;; adds the serial numbers to the crl
         new-full-crl (utils/generate-crl
-                       (.getIssuerX500Principal cacert)
+                       ;; ensure the issuer is identical
+                       (.getIssuerX500Principal crl)
```

Comment:
The "issuer" for the `cacert` ends up being the root cert.  In this case, we want to use the same issuer as the original CRL, so it matches.